### ISSUE TITLE
Flaky CompleteProcessInstanceAfterLeaderChangeTest.shouldCompleteProcessInstanceAfterLeaderChangeWithSnapshot

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#10561]
**Summary**

> Try to answer the following as best as possible

- How often does the test fail? Once so far
- Does it block your work? No
- Do we suspect that it is a real failure? Maybe

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Expecting actual not to be null
	at io.camunda.zeebe.it.util.ZeebeAssertHelper.assertElementCompleted(ZeebeAssertHelper.java:156)
	at io.camunda.zeebe.it.util.ZeebeAssertHelper.assertProcessInstanceCompleted(ZeebeAssertHelper.java:113)
	at io.camunda.zeebe.it.util.ZeebeAssertHelper.assertProcessInstanceCompleted(ZeebeAssertHelper.java:108)
	at io.camunda.zeebe.it.clustering.CompleteProcessInstanceAfterLeaderChangeTest.shouldCompleteProcessInstanceAfterLeaderChangeWithSnapshot(CompleteProcessInstanceAfterLeaderChangeTest.java:191)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

https://ci.zeebe.camunda.cloud/job/camunda/job/zeebe/job/main/862/

**Comment by Zelldon** [comment]
Might be related another fail on jenkins (but other method)

CompleteProcessInstanceAfterLeaderChangeTest.shouldCompleteInstanceAfterLeaderChange

https://ci.zeebe.camunda.cloud/job/camunda/job/zeebe/job/main/907/consoleFull


```
[ERROR] CompleteProcessInstanceAfterLeaderChangeTest.shouldCompleteInstanceAfterLeaderChange  Time elapsed: 34.951 s  <<< ERROR!
02:28:59 io.camunda.zeebe.client.api.command.ClientStatusException: deadline exceeded after 9.999554354s. [closed=[], open=[[buffered_nanos=1584955, remote_addr=0.0.0.0/127.0.0.1:7509]]]
02:28:59 	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.transformExecutionException(ZeebeClientFutureImpl.java:93)
02:28:59 	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:50)
02:28:59 	at io.camunda.zeebe.it.util.GrpcClientRule.deployProcess(GrpcClientRule.java:189)
02:28:59 	at io.camunda.zeebe.it.clustering.CompleteProcessInstanceAfterLeaderChangeTest.lambda$actions$5(CompleteProcessInstanceAfterLeaderChangeTest.java:108)
02:28:59 	at io.camunda.zeebe.it.clustering.CompleteProcessInstanceAfterLeaderChangeTest.shouldCompleteInstanceAfterLeaderChange(CompleteProcessInstanceAfterLeaderChangeTest.java:151)
02:28:59 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
02:28:59 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
02:28:59 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
02:28:59 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
02:28:59 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
02:28:59 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
02:28:59 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
02:28:59 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
02:28:59 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
02:28:59 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
02:28:59 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
02:28:59 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
02:28:59 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
02:28:59 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
02:28:59 	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
02:28:59 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
02:28:59 	at java.base/java.lang.Thread.run(Thread.java:833)
02:28:59 Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED: deadline exceeded after 9.999554354s. [closed=[], open=[[buffered_nanos=1584955, remote_addr=0.0.0.0/127.0.0.1:7509]]]
02:28:59 	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
02:28:59 	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
02:28:59 	at io.camunda.zeebe.client.impl.ZeebeClientFutureImpl.join(ZeebeClientFutureImpl.java:48)
02:28:59 	... 20 more
02:28:59 Caused by: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED: deadline exceeded after 9.999554354s. [closed=[], open=[[buffered_nanos=1584955, remote_addr=0.0.0.0/127.0.0.1:7509]]]
02:28:59 	at io.grpc.Status.asRuntimeException(Status.java:535)
02:28:59 	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:487)
02:28:59 	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:470)
02:28:59 	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:434)
02:28:59 	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:467)
02:28:59 	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:563)
02:28:59 	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
02:28:59 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:744)
02:28:59 	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:723)
02:28:59 	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
02:28:59 	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
02:28:59 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
02:28:59 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
02:28:59 	... 1 more
```

**Comment by megglos** [comment]
closing for now as not observed on GHA recently and also not occured on jenkins main builds for the past weeks since last failure
